app/src/main/java/com/minimalisticapps/priceconverter/data/remote/dto/BitPayExchangeRate.kt

```diff
@@ -4,24 +4,28 @@ import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
 import com.google.gson.annotations.SerializedName
+import java.math.BigDecimal
 
 @Entity(tableName = "Coins")
 data class BitPayExchangeRate(
     @ColumnInfo(name = "code")
     var code: String,
+
     @ColumnInfo(name = "rate")
     @SerializedName("value")
-    var rate: Double?,
+    var rate: BigDecimal?,
+
     @PrimaryKey(autoGenerate = false)
     @ColumnInfo(name = "name")
     val name: String,
+
     @ColumnInfo(name = "type")
     val type: String,
+
     @ColumnInfo(name = "unit")
     @SerializedName("unit")
     val unit: String,
+
     @ColumnInfo(name = "one_shit_coin_value")
-    var oneShitCoinValue: Double?,
-    @ColumnInfo(name = "one_shit_coin_value_string")
-    var oneShitCoinValueString: String
-)
\ No newline at end of file
+    var oneUnitOfShitcoinInBTC: BigDecimal?,
```

Comment:
BigDecimal was used in the original implementation for a reason. We work with money. It is necessary to use fixed decimal format and avoid a float/double precision problems.